revision 1.24
date: 2011/11/17 22:06:25;  author: jdennis%redhat.com;  state: Exp;  lines: +104 -104
fix compile time bug for SECKEYRSAPSSParams, only appears when NSS
version is >= 3.13

Index: py_nss.c
===================================================================
RCS file: /cvsroot/mozilla/security/python/nss/src/py_nss.c,v
retrieving revision 1.23
retrieving revision 1.24
diff -u -p -d -w -r1.23 -r1.24
--- a/src/py_nss.c	6 Jun 2011 15:21:13 -0000	1.23
+++ b/src/py_nss.c	17 Nov 2011 22:06:25 -0000	1.24
@@ -4669,7 +4669,7 @@ RSAPSSParams_format_lines(SECItem *item,
     }
 
     if (params.hashAlg) {
-        obj = oid_secitem_to_pystr_desc(&params->hashAlg->algorithm);
+        obj = oid_secitem_to_pystr_desc(&params.hashAlg->algorithm);
     } else {
         obj = PyString_FromString("default, SHA-1");
     }
@@ -4677,8 +4677,8 @@ RSAPSSParams_format_lines(SECItem *item,
     Py_CLEAR(obj);
 
     if (params.maskAlg) {
-        obj = oid_secitem_to_pystr_desc(&params->maskAlg->algorithm);
-        if (SEC_QuickDERDecodeItem(pool, &mask_hash_alg,
+        obj = oid_secitem_to_pystr_desc(&params.maskAlg->algorithm);
+        if (SEC_QuickDERDecodeItem(arena, &mask_hash_alg,
                                    SEC_ASN1_GET(SECOID_AlgorithmIDTemplate),
                                    &params.maskAlg->parameters) == SECSuccess) {
             obj1 = oid_secitem_to_pystr_desc(&mask_hash_alg);
